saleor/plugins/avatax/__init__.py

```diff
@@ -360,24 +360,29 @@ def get_order_lines_data(
             amount=undiscounted_amount,
         )
 
-        if undiscounted_amount != price_with_discounts_amount:
+        # for invoice transaction we want to include only undiscounted price
+        if (
+            not invoice_transaction_type
+            and undiscounted_amount != price_with_discounts_amount
+        ):
             append_line_to_data(
                 **append_line_to_data_kwargs,
                 amount=price_with_discounts_amount,
                 ref1=line.variant.sku,
             )
 
-    discount_amount = get_total_order_discount(order)
-    if discount_amount:
-        append_line_to_data(
-            data=data,
-            quantity=1,
-            amount=discount_amount.amount * -1,
-            tax_code=COMMON_DISCOUNT_VOUCHER_CODE,
-            item_code="Voucher",
-            name="Order discount",
-            tax_included=True,  # Voucher should be always applied as a gross amount
-        )
+    if not invoice_transaction_type:
```

Comment:
Are you sure of this? There is still a possibility that we add a voucher to the entire order. This voucher will be included here. In previous approach we send this to avatax (which was OK)

Reply:
Good point.